uaclient/files/state_files.py

```diff
@@ -232,3 +232,44 @@ def __init__(
 
 
 reboot_cmd_marker_file = UAFile("marker-reboot-cmds-required")
+
+
+class AnboxCloudCredentials(DataObject):
+    fields = [
+        Field("anbox_images_url", StringDataValue),
+        Field("anbox_images_resource_token", StringDataValue),
+        Field("anbox_cloud_apt_token", StringDataValue),
+        Field("anbox_cloud_apt_url", StringDataValue),
+    ]
+
+    def __init__(
+        self,
+        anbox_images_url: str,
+        anbox_images_resource_token: str,
+        anbox_cloud_apt_token: str,
+        anbox_cloud_apt_url: str,
+    ):
+        self.anbox_images_url = anbox_images_url
+        self.anbox_images_resource_token = anbox_images_resource_token
+        self.anbox_cloud_apt_token = anbox_cloud_apt_token
+        self.anbox_cloud_apt_url = anbox_cloud_apt_url
+
+
+class AnboxCloudData(DataObject):
+    fields = [
+        Field("anbox_attributes", AnboxCloudCredentials),
+    ]
+
+    def __init__(self, anbox_attributes: AnboxCloudCredentials):
+        self.anbox_attributes = anbox_attributes
```

Comment:
Do we expect anything else to need to go at this layer? If not, we could flatten the file and have the attributes at the top level

Reply:
Yes, right now there isn't. My concern is just that this might happen in the future, but maybe it is not that problematic

Reply:
+1